src/Pos/CLI.hs

```diff
@@ -283,17 +265,40 @@ walletPortOption portNum help =
         <> Opt.value portNum
         <> Opt.showDefault
 
-ipPortOption :: NetworkAddress -> Opt.Parser NetworkAddress
-ipPortOption na =
+externalNetworkAddressOption :: Maybe NetworkAddress -> Opt.Parser NetworkAddress
+externalNetworkAddressOption na =
```

Comment:
Yes when we spoke about this earlier I knew I would need the same change. I'll sort out the conflicts after your patch lands in master.